vyper/semantics/analysis/utils.py

```diff
@@ -646,10 +646,20 @@ def check_modifiability(node: vy_ast.VyperNode, modifiability: Modifiability) ->
 
     if isinstance(node, vy_ast.Call):
         args = node.args
+
+        # structs
         if len(args) == 1 and isinstance(args[0], vy_ast.Dict):
             return all(check_modifiability(v, modifiability) for v in args[0].values)
 
         call_type = get_exact_type_from_node(node.func)
+
+        # interfaces
+        from vyper.semantics.types.module import InterfaceT
```

Comment:
i don't really like the import cycle here, it would be best if there were enough annotation introduced during constant folding so that we didn't need to do this.